cli/src/commands/init.rs

```diff
@@ -53,13 +55,21 @@ pub(crate) fn cmd_init(
         .and_then(|_| wc_path.canonicalize())
         .map_err(|e| user_error_with_message("Failed to create workspace", e))?;
 
+    // Preserve existing behaviour where `jj init` is not able to create
+    // a colocated repo.
+    let colocated = false;
     if args.git || args.git_repo.is_some() {
-        git::git_init(ui, command, &wc_path, args.git_repo.as_deref())?;
+        git::git_init(ui, command, &wc_path, colocated, args.git_repo.as_deref())?;
+        writeln!(
+            ui.warning(),
+            "warning: `--git` and `--git-repo` are deprecated.
+Use `jj git init` to create git repos"
```

Comment:
Optional: Could be "Use `jj git init` instead".

Reply:
Makes sense. Done.